neo/Network/P2P/TaskManager.cs

```diff
@@ -266,5 +271,13 @@ internal protected override bool IsHighPriority(object message)
                     return false;
             }
         }
+
+        internal protected override bool ShallDrop(object message, IEnumerable queue)
+        {
+            if (!(message is TaskManager.NewTasks tasks)) return false;
+            // Remove duplicate tasks
+            if (queue.OfType<TaskManager.NewTasks>().Any(x => x.Payload.Type == tasks.Payload.Type && x.Payload.Hashes.SequenceEqual(tasks.Payload.Hashes))) return true;
```

Comment:
I think we should think more about this line.

Node A sends an `inv` for a block, and node B sends the same. In this case, the second message will be dropped. But if node A doesn't send `block`, we will stop sync.

Reply:
O.o, makes sense, @erikzhang. ahuehauhea We are destroying master2x...
Just kidding, I think it will be ok there for now! ahueahuea

Can we personalize this `IEnumerable queue` a little bit more? If we set timestamps for the `objects`. Maybe also some improvements could come on the management of `ProtocolHandler`

Reply:
Then maybe we need the duplicates, i think that with the FIFO's patch and the own messages patch, will be enought, it was tested with this both patches and without this one?

Reply:
I am testing a change for known hashes behaviour in ProtocolHandler related it, will share the code after the testing is done.

Reply:
Please check the PR #1024 which is for 2x.

Reply:
I still think we should remove this.

Reply:
So, let's remove this and open it in another PR.

I think that we added it before on master2.x. But we can separate it here and analyse in another moment.

Reply:
> I think we should think more about this line.
> 
> Node A sends an `inv` for a block, and node B sends the same. In this case, the second message will be dropped. But if node A doesn't send `block`, we will stop sync.

With `PendingHashes` it will not happen in a loop, since the hash will expire.